pixeltable/utils/azure_store.py

```diff
@@ -0,0 +1,297 @@
+            container_client = self.client().get_container_client(self.container_name)
+
+            for blob in blob_iterator:
+                # TODO: Figure out now to properly use batch method delete_blobs(), it doesn't seem to work properly
+                container_client.delete_blob(blob.name)
+                total_deleted += 1
+
+            # print(f"Deleted {total_deleted} blobs from container '{self.container_name}'.")
+            return total_deleted
+
+        except AzureError as e:
+            self.handle_azure_error(e, self.container_name, f'deleting with {self.prefix}')
+            raise
+
+    def list_objects(self, return_uri: bool, n_max: int = 10) -> list[str]:
+        """Return a list of objects found in the specified destination bucket.
+        Each returned object includes the full set of prefixes.
+        if return_uri is True, full URI's are returned; otherwise, just the object keys.
+        """
+        p = self.soa.prefix_free_uri if return_uri else ''
+        r: list[str] = []
+        try:
+            blob_iterator = self._get_filtered_blobs(tbl_id=None, tbl_version=None)
+            for blob in blob_iterator:
+                r.append(f'{p}{blob.name}')
+                if len(r) >= n_max:
+                    return r
+
+        except AzureError as e:
+            self.handle_azure_error(e, self.__container_name, f'list objects from {self.__base_uri}')
+        return r
+
+    @classmethod
+    def handle_azure_error(
+        cls, e: 'AzureError', container_name: str, operation: str = '', *, ignore_404: bool = False
+    ) -> None:
+        from azure.core.exceptions import ClientAuthenticationError, HttpResponseError, ResourceNotFoundError
+
+        if ignore_404 and isinstance(e, ResourceNotFoundError):
+            return
+
+        if isinstance(e, ResourceNotFoundError):
+            raise excs.Error(f'Container {container_name} or blob not found during {operation}: {str(e)!r}')
+        elif isinstance(e, ClientAuthenticationError):
+            raise excs.Error(f'Authentication failed for container {container_name} during {operation}: {str(e)!r}')
+        elif isinstance(e, HttpResponseError):
+            if e.status_code == 403:
+                raise excs.Error(f'Access denied to container {container_name} during {operation}: {str(e)!r}')
+            elif e.status_code == 412:
+                raise excs.Error(f'Precondition failed for container {container_name} during {operation}: {str(e)!r}')
+            else:
+                raise excs.Error(
+                    f'HTTP error during {operation} in container {container_name}: {e.status_code} - {str(e)!r}'
+                )
+        else:
+            raise excs.Error(f'Error during {operation} in container {container_name}: {str(e)!r}')
+
+    @classmethod
+    def create_az_raw(cls, endpoint_url: str, account_name: str | None = None, account_key: str | None = None) -> Any:
+        """Get a raw client without any locking"""
```

Comment:
What does 'raw' mean? Should this be create_client()? Why 'az'? (Refers to availability zone in AWS.)

Is there a reason for having it return Any?